plugins/plugin.go

```diff
@@ -156,13 +157,51 @@ func (p *Plugin) setEndpoints() error {
 	if p.Role == dcos.RoleAgent || p.Role == dcos.RoleAgentPublic {
 		p.Endpoints = []string{
 			"/system/v1/metrics/v0/node",
-			"/system/v1/metrics/v0/containers",
 		}
+
+		containers := []string{}
+		metricsURL := url.URL{
+			Scheme: p.MetricsProto,
+			Host:   net.JoinHostPort(p.MetricsHost, p.MetricsPort),
+			Path:   "/system/v1/metrics/v0/containers",
+		}
+
+		request := &http.Request{
+			Method: "GET",
+			URL:    &metricsURL,
+			Header: http.Header{
+				"Authorization": []string{fmt.Sprintf("token=%s", p.AuthToken)},
+			},
+		}
+
+		client := &http.Client{}
+		resp, err := client.Do(request)
+		if err != nil {
+			return err
+		}
+
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			logrus.Errorf("Encountered error reading response body, %s", err.Error())
+			return err
+		}
+
+		if err := json.Unmarshal(body, &containers); err != nil {
+			return err
+		}
+
+		for _, c := range containers {
```

Comment:
Looks like this is only fetched once at startup? What if containers are added/removed?